client/src/components/Avatar.js

```diff
@@ -19,39 +20,79 @@ const UploadImage = styled.button`
   width: 25px;
   margin-left: 35px;
   position: absolute;
+  align-items: center;
   justify-content: center;
+  background: url(${cameraIcon}) no-repeat;
+  background-position: center;
   border-radius: 50%;
   border: none;
   background-color: white;
   outline: none;
+  cursor: pointer;
   &:active {
     transform: translateY(1px) scale(0.9);
     box-shadow: 0px 5px 10px rgba(0, 0, 0, 20%);
   }
 `;
 
+const ImageInput = styled.input`
+  display: flex;
+  height: 20px;
+  width: 20px;
+  opacity: 0%;
+   position relative;
+   border-radius: 50%;
+  border: none;
+  focus {
+    outline: none;
+  }
+  overflow: hidden;
+`;
+
 const AvatarContainer = styled.div`
   display: flex;
-  width: 140px;
   height: auto;
   flex-direction: row-reverse;
   justify-content: center;
   align-items: flex-end;
 `;
-const Avatar = ({ onClick, src }) => {
+const Avatar = () => {
+  const { image, setImage } = useAvatar();
+
+  const handleImageUpload = async (event) => {
+    event.preventDefault();
+    const files = event.target.files;
+    const formData = new FormData();
+    formData.append('file', files[0]);
+    formData.append('upload_preset', 'bouybyw8');
```

Comment:
```suggestion
    formData.append('upload_preset', process.env.REACT_APP_CLOUDINARY_UPLOAD_PRESET);
```

https://create-react-app.dev/docs/adding-custom-environment-variables/

Reply:
I agree with you that this is not optimally solved. I tried it in the first step with a local.env file. Unfortunately, the upload via Cloudinary did not work. But I will have a look at this point again.